packages/core/test/unit/test-runner/RestartWorkerDecorator.spec.ts

```diff
@@ -0,0 +1,98 @@
+import { TestRunner2, DryRunOptions, MutantRunOptions, DryRunResult, MutantRunResult } from '@stryker-mutator/api/test_runner';
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { factory } from '@stryker-mutator/test-helpers';
+
+import TestRunnerDecorator from '../../../src/test-runner/TestRunnerDecorator';
+import RestartWorkerDecorator from '../../../src/test-runner/RestartWorkerDecorator';
+
+describe(RestartWorkerDecorator.name, () => {
+  let workerWithoutRestarts: RestartWorkerDecorator;
+  let workerWithRestarts: RestartWorkerDecorator;
+  let testRunner: sinon.SinonStubbedInstance<Required<TestRunner2>>;
+  const sandbox = sinon.createSandbox();
+
+  beforeEach(() => {
+    testRunner = factory.testRunner();
+    workerWithoutRestarts = new RestartWorkerDecorator(() => testRunner, { restartAfterRuns: 0 });
+    workerWithRestarts = new RestartWorkerDecorator(() => testRunner, { restartAfterRuns: 1 });
```

Comment:
Nit: Would personally like a `sut` per test. So would probably create the `sut` in each test (maybe with a `createSut` helper function).